File: tools/code-parser-sidecar/fixtures/py-sample/service.py
from util import format_name


class UserService:
    def __init__(self):
        self.users = []

    def add_user(self, user):
        self.validate(user)
        self.users.append(user)

    def validate(self, user):
        return format_name(user["name"]) != ""


def create_default_service():
    svc = UserService()
    svc.add_user({"name": "alkut"})
    return svc

File: tools/code-parser-sidecar/fixtures/ts-sample/src/service.ts
import { formatName } from "./util";

export interface User {
  id: number;
  name: string;
}

export class UserService {
  private users: User[] = [];

  addUser(user: User): void {
    this.validate(user);
    this.users.push(user);
  }

  validate(user: User): boolean {
    return user.id > 0 && formatName(user.name).length > 0;
  }

  findUser(id: number): User | undefined {
    return this.users.find((u) => u.id === id);
  }
}

export function createDefaultService(): UserService {
  const svc = new UserService();
  svc.addUser({ id: 1, name: "alkut" });
  return svc;
}
